components/index_weaviate/src/main.py

```diff
@@ -54,9 +55,13 @@ def __init__(
 
     def write(self, dataframe: dd.DataFrame) -> None:
         with self.client.batch as batch:
-            for part in dataframe.partitions:
+            for part in tqdm(
+                dataframe.partitions,
+                desc="Processing partitions",
+                total=dataframe.npartitions,
+            ):
                 df = part.compute()
-                for row in df.itertuples():
+                for row in tqdm(df.itertuples(), desc="Processing rows", total=len(df)):
```

Comment:
Have you tested this?
Providing progress on the row level might be a bit much. I guess partitions would be enough?

Reply:
Yeah indeed this just adds in another bar but the process time per partition is generally not that long, removed it